python/social_bot/envs/grocery_ground.py

```diff
@@ -222,11 +222,18 @@ def __init__(self,
 
         obs_data = self._get_observation()
         if self._use_image_obs:
-            obs_data_space = gym.spaces.Box(
-                low=0, high=255, shape=obs_data.shape, dtype=np.uint8)
+            obs_data_space = gym.spaces.Tuple([
+                gym.spaces.Box(
+                    low=0, high=1.0, shape=obs_data[0].shape, dtype=np.float32),
+                gym.spaces.Box(
+                    low=-np.inf,
+                    high=np.inf,
+                    shape=obs_data[1].shape,
+                    dtype=np.float32)
```

Comment:
if with_language is True, the observation will be nested. I think it's better not to have nested structure. A flat dict for all different fields will be easier to handle. So space.Dict(image=...., internal_state=..., sentense=...) is better.

Reply:
Also comment the observation space at the comment section of this class.

Reply:
> don't use float32, this will increase the memory consumption when using replay buffer.

should we uint8 instead, and change image of pr2 gripper to uint8 as well?

Reply:
Yes.

Reply:
> if with_language is True, the observation will be nested. I think it's better not to have nested structure. A flat dict for all different fields will be easier to handle. So space.Dict(image=...., internal_state=..., sentense=...) is better
Sure, it is better this way. I was thinking about to be the same as pr2 gripper. Should observation of pr2gripper also changed into this structure?

Reply:
pr2 gripper does not use language, so its observation is a flat dict.